File: c-004/main.hs
isPalindrome num =
    show num == reverse (show num)

largestPalindrome num
    | num == 0      = 0
    | otherwise     = max result (largestPalindrome (num - 1))
    where
        result = head [num * a| a <- reverse [0..num], isPalindrome (num * a)]

main =
    print $ largestPalindrome 999

File: c-004/main.py
def isPalindrome(num):
	strnum = str(num)
	return strnum == strnum[::-1]

def largestPalindrome(num):
    result = 0
    for i in range(num, 0, -1):
        for j in range(i, 0, -1):
            if isPalindrome(i * j):
                result = max(result, i * j)
                break
    return result

print(largestPalindrome(999))
